src/App.tsx

```diff
@@ -1,15 +1,47 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { DropDownMenu } from './components/DropdownMenu';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectPerson, setSelectPerson] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), []);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+
+    setSelectPerson(value);
+
+    if (value === '') {
+      setAppliedQuery('');
+    } else {
+      applyQuery(value);
+    }
```

Comment:
This part of code does not cover the case when the user entered spaces only. According to the checklist, the filter should not be called in this case. You should add a check for a string consisting only of spaces.